src/psbtv2.ts

```diff
@@ -358,8 +358,14 @@ export class PsbtV2 extends PsbtV2Maps {
   }
 
   set PSBT_GLOBAL_TX_VERSION(version: number) {
+    let workingVersion = version;
+    if (workingVersion < 2) {
+      console.warn(`PsbtV2 cannot have a global tx version less than 2. Version ${workingVersion} specified. Setting to version 2.`);
+      workingVersion = 2;
+    }
+
```

Comment:
I wanted to note here that I think we should be careful how we handle this. 

Just for background, here's the language from the [BIP](https://github.com/bitcoin/bips/blob/master/bip-0370.mediawiki#user-content-Constructor): 

It [says](https://github.com/bitcoin/bips/blob/master/bip-0370.mediawiki#cite_ref-3-0):
> The transaction version number must be set to at least 2.

To add more clarity, the reason for this is given:

> Why does the transaction version number need to be at least 2? The transaction version number is part of the validation rules for some features such as OP_CHECKSEQUENCEVERIFY. Since it is backwards compatible, and there are other ways to disable those features (e.g. through sequence numbers), it is easier to require transactions be able to support these features than to try to negotiate the transaction version number.

So the reason is essentially to make negotiation easier for special features. 

With this context, I think that changing the version in the setter is probably not the right approach. At the very least, I would say that if we think this is a hard rule that should be stuck to, then we should throw an error here rather than maleate the transaction essentially (this is particularly problematic if you're converting from V0). 

On the other hand, it seems like other implementations are not very opinionated on this and it also does break backwards compatibility in a sense which was another stated goal of BIP370. 

All that to say, I think I would argue for some way to at least allow for versions less than 2, at least in some narrow edge cases. Maybe we can accept an argument `allowLegacy: boolean` that allows a version to be set to < 2. If this is not `true` then the method will throw, and if it is true then a warning is logged but the version is set to the "wrong" version.